homeassistant/components/alarmdecoder/config_flow.py

```diff
@@ -0,0 +1,322 @@
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        if user_input is not None:
+            self.protocol = user_input[CONF_PROTOCOL]
+            return await self.async_step_protocol()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_PROTOCOL): vol.In(
+                        [PROTOCOL_SOCKET, PROTOCOL_SERIAL]
+                    ),
+                }
+            ),
+        )
+
+    async def async_step_protocol(self, user_input=None):
+        """Handle AlarmDecoder protocol setup."""
+        errors = {}
+        if user_input is not None:
+            connection = {}
+            if self.protocol == PROTOCOL_SOCKET:
+                baud = connection[CONF_DEVICE_BAUD] = DEFAULT_DEVICE_BAUD
+                host = connection[CONF_HOST] = user_input[CONF_HOST]
+                port = connection[CONF_PORT] = user_input[CONF_PORT]
+                title = f"{host}:{port}"
+                device = SocketDevice(interface=(host, port))
+            if self.protocol == PROTOCOL_SERIAL:
+                path = connection[CONF_DEVICE_PATH] = user_input[CONF_DEVICE_PATH]
+                baud = connection[CONF_DEVICE_BAUD] = user_input[CONF_DEVICE_BAUD]
+                title = path
+                device = SerialDevice(interface=path)
+
+            controller = AdExt(device)
+            try:
+                with controller:
+                    controller.open(baudrate=baud)
+                return self.async_create_entry(
+                    title=title, data={CONF_PROTOCOL: self.protocol, **connection}
+                )
+            except NoDeviceError:
+                errors["base"] = "service_unavailable"
+
+        if self.protocol == PROTOCOL_SOCKET:
+            schema = vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=DEFAULT_DEVICE_HOST): str,
+                    vol.Required(CONF_PORT, default=DEFAULT_DEVICE_PORT): int,
+                }
+            )
+        if self.protocol == PROTOCOL_SERIAL:
+            schema = vol.Schema(
+                {
+                    vol.Required(CONF_DEVICE_PATH, default=DEFAULT_DEVICE_PATH): str,
+                    vol.Required(CONF_DEVICE_BAUD, default=DEFAULT_DEVICE_BAUD): int,
+                }
+            )
+
+        return self.async_show_form(
+            step_id="protocol", data_schema=schema, errors=errors,
+        )
+
+
+class AlarmDecoderOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle AlarmDecoder options."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry):
+        """Initialize AlarmDecoder options flow."""
+        self.arm_options = config_entry.options.get(OPTIONS_ARM, DEFAULT_ARM_OPTIONS,)
```

Comment:
```suggestion
        self.arm_options = config_entry.options.get(OPTIONS_ARM, DEFAULT_ARM_OPTIONS)
```

Reply:
done in https://github.com/home-assistant/core/pull/37998/commits/7566b8a722ae489b073c23ff8eb32fe4e320523c :+1: